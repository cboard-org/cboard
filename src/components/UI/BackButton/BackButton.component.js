import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';

import IconButton from '../IconButton';
import messages from './BackButton.messages';
import './BackButton.css';

const propTypes = {
  intl: intlShape.isRequired,
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

function BackButton({ intl, disabled, onClick }) {
  const backString = intl.formatMessage(messages.back);

  return (
    <IconButton label={backString} disabled={disabled} onClick={onClick}>
      <span className="BackButton__icon">
        <ArrowBackIcon />
      </span>
    </IconButton>
  );
}

BackButton.propTypes = propTypes;

export default injectIntl(BackButton);
