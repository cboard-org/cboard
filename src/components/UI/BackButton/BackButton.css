.BackButton__icon {
  display: flex;
}

[dir='rtl'] .BackButton__icon {
  transform: scale(-1);
}
