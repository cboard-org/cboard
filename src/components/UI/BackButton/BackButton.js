import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import { withStyles } from '@material-ui/core/styles';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import ArrowForwardIcon from '@material-ui/icons/ArrowForward';

import IconButton from '../IconButton';
import messages from './BackButton.messages';

const propTypes = {
  /**
   * If true, back button is disabled
   */
  disabled: PropTypes.bool,
  /**
   * @ignore
   */
  intl: intlShape.isRequired,
  /**
   * @ignore
   */
  theme: PropTypes.object.isRequired,
  /**
   * Callback fired when back button is clicked
   */
  onClick: PropTypes.func.isRequired
};

function BackButton(props) {
  const { intl, theme, ...rest } = props;
  const label = intl.formatMessage(messages.back);

  return (
    <IconButton label={label} {...rest}>
      {theme.direction === 'ltr' ? <ArrowBackIcon /> : <ArrowForwardIcon />}
    </IconButton>
  );
}

BackButton.propTypes = propTypes;

export default withStyles(null, { withTheme: true })(injectIntl(BackButton));
