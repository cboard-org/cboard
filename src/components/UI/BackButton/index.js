export { default } from './BackButton.component';
