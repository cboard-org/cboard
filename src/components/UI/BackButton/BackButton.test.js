import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';

import BackButton from './BackButton';

const intlMock = { formatMessage: () => 'dummy message' };

describe('BackButton tests', () => {
  test('default renderer', () => {
    const props = {
      intl: intlMock,
      theme: {},
      onClick: () => {}
    };
    shallowMatchSnapshot(<BackButton {...props} />);
  });
});
