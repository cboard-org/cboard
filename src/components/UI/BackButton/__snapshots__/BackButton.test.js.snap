// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton tests default renderer 1`] = `
<BackButton
  classes={Object {}}
  intl={
    Object {
      "formatMessage": [Function],
    }
  }
  onClick={[Function]}
  theme={Object {}}
/>
`;
